restapi/auth.go

```diff
@@ -72,37 +73,62 @@ func (api api) AuthCallback(w http.ResponseWriter, r *http.Request) render.Rende
 	// redirect if invalid user
 	err = api.database.FirstOrCreate(&user, models.User{Email: user.Email}).Error
 	if err != nil {
-		http.Redirect(w, r, "/api/unauthorized", http.StatusTemporaryRedirect)
+		http.Redirect(w, r, "/unauthorized", http.StatusTemporaryRedirect)
 	}
 
-	// if valid create jwt token
-	jwtToken, err := generateJWTToken(user.Email)
+	// if valid set jwt token in cookie
+	err = generateJWTToken(user.Email, w)
 	if err != nil {
 		return rest.ErrInternal(api.logger, err)
 	}
 
-	return rest.JSONStatusOK(jwtToken)
+	return rest.JSONStatusOK("Authenticated successfully")
 }
 
-func generateJWTToken(email string) (string, error) {
+func generateJWTToken(email string, w http.ResponseWriter) error {
 	jwtExpiry, err := config.GetTokenExpiryDuration()
 	if err != nil {
-		return "", err
+		return err
 	}
+	expiration := time.Now().Add(jwtExpiry)
 
 	claim := &models.Claims{
 		Email: email,
 		StandardClaims: jwt.StandardClaims{
-			ExpiresAt: time.Now().Add(jwtExpiry).Unix(),
+			ExpiresAt: expiration.Unix(),
 			Issuer:    config.GetTokenIssuer(),
 		},
 	}
 
 	token := config.GetJWTKey()
 	_, signedToken, err := token.Encode(claim)
 	if err != nil {
-		return "", err
+		return err
 	}
 
-	return signedToken, nil
+	secure := (config.GetMode() == config.MODE_PROD)
```

Comment:
what does this do?

Reply:
if in prod, `Secure` field is set to true in cookie: this makes sure that cookie is only included in reqs over https

Reply:
this may be an issue since we don't have https right now in prod